### PR TITLE
[MERGE] Merge branch 1.4 to master

### PR DESCRIPTION
**Pull request by calvinjia**
Merge from branch-1.4 to master. Regenerated proto files with protoc version 2.5.0.

### CODE REVIEW

File: core/server/pom.xml [hunk 1]
```diff
@@ -35,14 +35,19 @@
       <groupId>com.google.protobuf</groupId>
       <artifactId>protobuf-java</artifactId>
     </dependency>
+    <!-- alluxio-core-common should come before alluxio-core-client as alluxio-core-client shade some dependencies
```
Comment (jsimsa) [inline core/server/pom.xml]: I don't think this change is needed anymore

Comment (calvinjia) [inline core/server/pom.xml]: I'll remove it here and later in 1.4

File: core/server/pom.xml [hunk 2]
```diff
@@ -35,11 +35,6 @@
       <groupId>com.google.protobuf</groupId>
       <artifactId>protobuf-java</artifactId>
     </dependency>
-    <!-- alluxio-core-common should come before alluxio-core-client as alluxio-core-client shade some dependencies
```
Comment (jsimsa) [inline core/server/pom.xml]: pei acutally took care of this in: https://github.com/Alluxio/alluxio/pull/4570

Comment (jsimsa) [inline core/server/pom.xml]: so if you remove this commit and simply merge branch-1.4, it should be all good

**Pull request merged**